src/librustc/traits/error_reporting.rs

```diff
@@ -1050,6 +1050,29 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
         if let Some(found_span) = found_span {
             err.span_label(found_span, format!("takes {}", found_str));
 
+            // Suggest to take and ignore the arguments with expected_args_length `_`s if
+            // found arguments is empty(Suppose the user just wants to ignore args in this case).
+            // like `|_, _|` for closure with 2 expected args.
+            if found_args.is_empty() && is_closure {
+                let mut underscores = "_".repeat(expected_args.len())
+                                      .split("")
+                                      .filter(|s| !s.is_empty())
+                                      .collect::<Vec<_>>()
+                                      .join(", ");
+                err.span_suggestion_with_applicability(
+                    found_span,
+                    &format!("change the closure to take and ignore the expected argument{}",
```

Comment:
_Nitpick_: I feel that this would read better as "you can change the closure...". That way we don't imply to the user that this is what they _must_ do, only something they _can_ do.